### PR TITLE
feat: node.js package release flow

### PR DESCRIPTION
**Pull request by borcherd**
Changelog:
1. There is one workflow (pr-test.yml) which handles testing when a PR is created. This workflow will:
- run the basic tests, they need to pass else it'll show a red cross next to the action indicating that the tests failed
- run the live tests, they do NOT need to pass. If the tests fail, it'll post a comment on the PR saying 'live tests failed, please check the logs'

2. These is one workflow (npm-publish.yml) which handles publishing a new sdk version to npm when a github release is created. 
   
Please add the following github secrets: GH_TOKEN (https://github.com/settings/tokens), ETHERSCAN_API_KEY, INFURA_GOERLI_RPC, TEST_WALLET_PRIVATE_KEY & NPM_TOKEN

Further improvements would be to improve the testing suite. We should clean up & make the live tests also required to pass.

### CODE REVIEW

**Comment by Hugo0** [comment]
2 things:
- mixing concerns in PR (tests renaming + github automation)
- I don't get why some of these tests are moved. e.g. `createClaimXChainPayload.test.ts` is not state changing

File: .github/workflows/npm-publish.yml
Comment (Hugo0) [inline .github/workflows/npm-publish.yml:1]: image
Is there any check that the tests all pass? Or what happens if one fails?

Is the status of the tests displayed after running?

**Comment by borcherd** [comment]
Tests are failing because the secrets need to be defined in the repository

**Comment by Hugo0** [comment]
image
why can i merge even though the checks haven't been completed yet?

**Comment by Hugo0** [comment]
image
added all secrets. rest lgtm. nice job :)

**Pull request merged**